### PR TITLE
Improve bonus points apply validation

### PR DESCRIPTION
**Pull request by bartektw**

### CODE REVIEW

File: src/Validator/Constraints/BonusPointsApplyValidator.php [hunk 1]
```diff
@@ -40,11 +40,19 @@ public function validate($bonusPoints, Constraint $constraint): void
...
         $eligibleBonusPointsStrategies = $this->extractOnlyEligibleStrategies($bonusPointsStrategies);
 
+        if (\count($eligibleBonusPointsStrategies) === 0) {
+            $this->context->buildViolation($constraint->messageInvalidOrderItem)->addViolation();
+        }
+
         foreach ($eligibleBonusPointsStrategies as $eligibleBonusPointsStrategy) {
             if ($bonusPoints % 100 !== 0 || $bonusPoints < 100) {
-                $this->context->buildViolation($constraint->message)->addViolation();
+                $this->context->buildViolation($constraint->messageInvalidNumber)->addViolation();
```
Comment (senghe) [inline src/Validator/Constraints/BonusPointsApplyValidator.php]: maybe better name would be invalidStrategyPointsValueMessage?

File: src/Validator/Constraints/BonusPointsApplyValidator.php [hunk 2]
```diff
@@ -40,11 +40,19 @@ public function validate($bonusPoints, Constraint $constraint): void
...
         $bonusPointsStrategies = $this->bonusPointsStrategyRepository->findActiveByCalculatorType(PerOrderPriceCalculator::TYPE);
 
+        if (\count($bonusPointsStrategies) === 0) {
+            return;
+        }
+
         $eligibleBonusPointsStrategies = $this->extractOnlyEligibleStrategies($bonusPointsStrategies);
 
+        if (\count($eligibleBonusPointsStrategies) === 0) {
+            $this->context->buildViolation($constraint->messageInvalidOrderItem)->addViolation();
```
Comment (senghe) [inline src/Validator/Constraints/BonusPointsApplyValidator.php]: maybe better name would be invalidStrategyPointsCountMessage?

File: src/Validator/Constraints/BonusPointsApplyValidator.php [hunk 3]
```diff
@@ -40,11 +40,19 @@ public function validate($bonusPoints, Constraint $constraint): void
     {
         $bonusPointsStrategies = $this->bonusPointsStrategyRepository->findActiveByCalculatorType(PerOrderPriceCalculator::TYPE);
 
+        if (\count($bonusPointsStrategies) === 0) {
```
Comment (senghe) [inline src/Validator/Constraints/BonusPointsApplyValidator.php:43]: I see no test checking this condition

File: src/Validator/Constraints/BonusPointsApplyValidator.php [hunk 4]
```diff
@@ -40,11 +40,19 @@ public function validate($bonusPoints, Constraint $constraint): void
...
+            return;
+        }
+
         $eligibleBonusPointsStrategies = $this->extractOnlyEligibleStrategies($bonusPointsStrategies);
 
+        if (\count($eligibleBonusPointsStrategies) === 0) {
```
Comment (senghe) [inline src/Validator/Constraints/BonusPointsApplyValidator.php:49]: I see no test checking this condition

File: src/Validator/Constraints/BonusPointsApply.php [hunk 5]
```diff
@@ -9,7 +9,10 @@
 final class BonusPointsApply extends Constraint
 {
     /** @var string */
-    public $message;
+    public $messageInvalidNumber;
```
Comment (senghe) [inline src/Validator/Constraints/BonusPointsApply.php]: Why both of this fields have no value assigned? I don't see them assigned in the entire class

**Review by senghe — CHANGES_REQUESTED** [review]

File: src/Validator/Constraints/BonusPointsApplyValidator.php (hunk 1 above)
Comment (bartektw) [reply src/Validator/Constraints/BonusPointsApplyValidator.php]: I think, better name would be invalidBonusPointsValueMessage

File: src/Validator/Constraints/BonusPointsApplyValidator.php (hunk 2 above)
Comment (bartektw) [reply src/Validator/Constraints/BonusPointsApplyValidator.php]: I think, better name would be noProductsFromBonusPointsStrategyTaxonMessage

File: src/Validator/Constraints/BonusPointsApplyValidator.php (hunk 1 above)
Comment (senghe) [reply src/Validator/Constraints/BonusPointsApplyValidator.php]: 👍

File: src/Validator/Constraints/BonusPointsApplyValidator.php (hunk 2 above)
Comment (senghe) [reply src/Validator/Constraints/BonusPointsApplyValidator.php]: 👍

**Review by senghe — APPROVED** [review]

**Pull request merged**